internal/provider/ovirt/base.go

```diff
@@ -0,0 +1,48 @@
+package ovirt
+
+import (
+	"fmt"
+
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/internal/provider"
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+)
+
+//
+type ovirtProvider struct {
+	Log logrus.FieldLogger
+}
+
+// NewOvirtProvider creates a new oVirt provider.
+func NewOvirtProvider(log logrus.FieldLogger) provider.Provider {
+	return &ovirtProvider{
+		Log: log,
+	}
+}
+
+// Name returns the name of the provider
+func (p *ovirtProvider) Name() models.PlatformType {
+	return models.PlatformTypeOvirt
+}
+
+func (p *ovirtProvider) IsHostSupported(host *models.Host) (bool, error) {
+	hostInventory, err := common.UnmarshalInventory(host.Inventory)
+	if err != nil {
+		return false, fmt.Errorf("error marshaling host to inventory, error %w", err)
```

Comment:
prefer to use errors.Wrapf

Reply:
We usually use errors.New on new error we want to trow and fmt.errorf with a %w format on wrapped errors.
See for a comparison:
- https://stackoverflow.com/questions/61933650/whats-the-difference-between-errors-wrapf-errors-errorf-and-fmt-errorf
- https://go.dev/blog/go1.13-errors

Anyway I can go either way, its just the way we did it through the project

Reply:
@filanov are you ok with maintaining the current code?

Reply:
yes